### PR TITLE
Validate grpc and pbnj authority strings for tinkerbell

### PR DESCRIPTION
**Pull request by chrisdoherty4**
Validate grpc and pbnj authority strings in the Tinkerbell cluster configuration read from disk.

The grpc and pbnj authority strings are addresses to a running instance of Tinkerbell PBnJ and Tink services. This PR validates the strings upon reading them in.

### CODE REVIEW

File: pkg/providers/tinkerbell/validator.go [hunk 1]
```diff
@@ -245,3 +254,17 @@ func sumWorkerNodeCounts(nodes []v1alpha1.WorkerNodeGroupConfiguration) int {
...
+// validateAddressWithPort validates address is a hostname:port combination. The port is required.
+func validateAddressWithPort(address string) error {
+	host, _, err := net.SplitHostPort(address)
+	if err != nil {
+		return err
+	}
+
+	if host == "" {
+		return fmt.Errorf("missing host: %v", address)
+	}
```
[the remark this answers is not in the document]
Comment (chrisdoherty4) [reply pkg/providers/tinkerbell/validator.go:267]: `SplitHostPort` does that. It throws an error if its missing.

File: pkg/providers/tinkerbell/validator.go [hunk 2]
```diff
@@ -245,3 +254,17 @@ func sumWorkerNodeCounts(nodes []v1alpha1.WorkerNodeGroupConfiguration) int {
 	}
 	return requestedNodesCount
 }
+
+// validateAddressWithPort validates address is a hostname:port combination. The port is required.
+func validateAddressWithPort(address string) error {
+	host, _, err := net.SplitHostPort(address)
```
Comment (abhinavmpandey08) [inline pkg/providers/tinkerbell/validator.go:260]: we should validate the port as well maybe using this https://github.com/aws/eks-anywhere/blob/main/pkg/networkutils/networkutils.go#L9-L12

File: pkg/providers/tinkerbell/validator_test.go [hunk 3]
```diff
@@ -56,6 +60,77 @@ func TestValidateMinimumRequiredTinkerbellHardware_EtcdUnspecified(t *testing.T)
...
+	err := validator.ValidateTinkerbellConfig(context.Background(), datacenter)
+
+	assert.NoError(t, err)
+}
+
+func TestValidateTinkerbellConfig_InvalidGrpcAuthority(t *testing.T) {
+	cases := map[string]string{
+		"Missing port":     "1.1.1.1",
+		"Missing hostname": ":44",
+	}
```
Comment (abhinavmpandey08) [inline pkg/providers/tinkerbell/validator_test.go:84]: should we add a case where the port is provided but invalid, like `1.1.1.1:abc` or `1.1.1.1:111111`

File: pkg/providers/tinkerbell/validator_test.go [hunk 4]
```diff
@@ -75,3 +150,15 @@ func newHardwareConfigWithHardware(hardwareCount int) hardware.HardwareConfig {
 		Hardwares: make([]tinkv1alpha1.Hardware, hardwareCount),
 	}
 }
+
+func newValidTinkerbellDatacenterConfig() *v1alpha1.TinkerbellDatacenterConfig {
+	return &v1alpha1.TinkerbellDatacenterConfig{
+		Status: v1alpha1.TinkerbellDatacenterConfigStatus{},
+		Spec: v1alpha1.TinkerbellDatacenterConfigSpec{
+			TinkerbellIP:           "1.1.1.1",
+			TinkerbellCertURL:      "1.1.1.1",
```
Comment (abhinavmpandey08) [inline pkg/providers/tinkerbell/validator_test.go:159]: typically the cert url is of the format `http://hostname:port/cert`

File: pkg/providers/tinkerbell/validator.go (hunk 2 above)
Comment (chrisdoherty4) [reply pkg/providers/tinkerbell/validator.go]: Done.

File: pkg/providers/tinkerbell/validator_test.go (hunk 4 above)
Comment (chrisdoherty4) [reply pkg/providers/tinkerbell/validator_test.go]: Done.

File: pkg/providers/tinkerbell/validator.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (chrisdoherty4) [reply pkg/providers/tinkerbell/validator.go:269]: I think this is resolved with additional checks from https://github.com/aws/eks-anywhere/pull/1321#discussion_r816466035. Feel free to reopen.

**Review by abhinavmpandey08 — APPROVED** [review]
/lgtm

**Pull request merged**